angular/src/shared/handlers/atomic-swap-key-handler.ts

```diff
@@ -21,22 +23,38 @@ export class AccountBalanceHandler implements ActionHandler {
       throw Error('The params walletId and accountId are missing.');
     }
 
+    if (params.walletId.includePrivateKey) {
+      // whould we ask for consent when fetching the swap private key?
+    }
+
     return {
       content: state.message.request.params[0],
-      consent: false,
+      consent: false
     };
   }
 
   async execute(state: ActionState, permission: Permission): Promise<ActionResponse> {
+
     var data = state.content as any;
 
     // get the account
-    const { network, account, accountState, accountHistory } = await this.backgroundManager.getAccount(data.walletId, data.accountId);
-
-    if (state.content) {
-      return { key: data.key, content: { balance: accountHistory.balance }, network: network.id };
-    } else {
-      return { key: '', signature: '', content: null, network: network.id };
+    const { network, node } = await this.backgroundManager.getDerivedKeyFromWalletPath(data.walletId, data.accountId, "0'/0");
+
+    if (state.content)
+    {
+      var response;
+
+      if (data.includePrivateKey) {
+        response = { publicKey: this.utility.keyToHex(node.publicKey), privateKey:  this.utility.keyToHex(node.privateKey) };
```

Comment:
this may also be problematic and we should discuss weather its ok to fetch the private key of a specific feature

Reply:
No, it's not acceptable :-)